src/polytopes/chain.jl

```diff
@@ -33,6 +33,17 @@ function vertices(c::Chain)
   end
 end
 
+"""
+    segments(chain)
+
+Return the segments linking consecutive points of the `chain`.
+"""
+function segments(c::Chain)
+  vs = vertices(c)
+  n = length(vs)
+  [Segment(view(vs, [i,i+1])) for i in 1:n-1]
```

Comment:
We could return a generator instead of materializing it. Just replace the brackets by parenthesis.